kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -278,25 +281,31 @@ private CallSummaryRecord aggregateSummaryRecord(CallSummaryRecordDto csr) {
     @MotechListener(subjects = { PROCESS_SUMMARY_RECORD_SUBJECT })
     public void processCallSummaryRecord(MotechEvent event) {
 
-        CallSummaryRecordDto csr = (CallSummaryRecordDto) event.getParameters().get(CSR_PARAM_KEY);
-        String subscriptionId = csr.getRequestId().getSubscriptionId();
-        CallSummaryRecord record = aggregateSummaryRecord(csr);
+        CallSummaryRecordDto csrDto = (CallSummaryRecordDto) event.getParameters().get(CSR_PARAM_KEY);
+        String subscriptionId = csrDto.getRequestId().getSubscriptionId();
+        CallSummaryRecord csr = aggregateSummaryRecord(csrDto);
 
         CallRetry callRetry = callRetryDataService.findBySubscriptionId(subscriptionId);
         Subscription subscription = subscriptionDataService.findBySubscriptionId(subscriptionId);
 
 
-        if (csr.getFinalStatus() == FinalCallStatus.SUCCESS) {
-            completeSubscriptionIfNeeded(subscription, record, callRetry);
-        }
+        switch (csrDto.getFinalStatus()) {
+            case SUCCESS:
+                completeSubscriptionIfNeeded(subscription, csr, callRetry);
+                break;
 
-        if (csr.getFinalStatus() == FinalCallStatus.FAILED) {
-            rescheduleCall(subscription, record, callRetry);
-        }
+            case FAILED:
+                rescheduleCall(subscription, csr, callRetry);
+                break;
+
+            case REJECTED:
+                deactivateSubscription(subscription, callRetry);
+                break;
 
-        if (csr.getFinalStatus() == FinalCallStatus.REJECTED) {
-            deactivateSubscription(subscription, callRetry);
+            default:
+                throw new IllegalStateException(String.format("Invalid FinalCallStatus: %s", csrDto.getFinalStatus()));
```

Comment:
Its probably better to just log/alert and fail here since an exception on the handler puts the event back on the queue to be picked up by someone else.